Работаем с датой

M2rk13 opened the pull request:
Функция **generate_random_date** заменена на **mt_rand** с указанием ранее начала диапазона случайной временной точки.
Замена произведена т.к. функция **generate_random_date** генерирует разные даты с одинаковым временем, равным текущему.

---
:mortar_board: [Работаем с датой](https://up.htmlacademy.ru/php/12/user/1415187/tasks/8)

M2rk13 committed:
Merge pull request #4 from M2rk13/module2-task2

Module2 task2

M2rk13 committed:
Модуль 3, завершено задание по работе со временем.

M2rk13 committed:
Добавлены правки для улучшения читаемости кода

M2rk13 committed:
Merge pull request #5 from M2rk13/module3-task1

Module3 task1

M2rk13 committed:
Merge branch 'master' into module3-task2

M2rk13 committed:
Update functions.php

заменен проблемный код,
внесены небольшие улучшения по оформлению кода

M2rk13 committed:
Update functions.php

00dev commented on templates/main.php:
```diff
@@ -141,7 +141,8 @@
                             <div class="post__info">
                                 <b class="post__author-name"><?=htmlspecialchars($post['author'])?></b>
-                                <time class="post__time" datetime="">дата</time>
+                                <?php $post_time = random_date_func () ?>
```
Используйте все-таки `generate_random_date()` Не забывайте передавать индекс поста (это позволяет формировать разные даты и время)

00dev commented on templates/main.php:
```diff
@@ -141,7 +141,8 @@
                                 <b class="post__author-name"><?=htmlspecialchars($post['author'])?></b>
-                                <time class="post__time" datetime="">дата</time>
+                                <?php $post_time = random_date_func () ?>
+                                <time class="post__time" datetime="<?=gmdate('Y-m-d H:i:s', $post_time)?>"><?=time_difference($post_time)?></time>
```
Не хватает атрибута `title`

M2rk13 committed:
Улучшена реализация согласно заданию

Заменена собственная функция на более подходящую из файла helpers.php
Тегу <time> добавлен атрибут title для отображения даты по наведению

00dev commented on templates/main.php:
```diff
@@ -141,7 +141,8 @@
                                 <b class="post__author-name"><?=htmlspecialchars($post['author'])?></b>
-                                <time class="post__time" datetime="">дата</time>
+                                <?php $post_time = generate_random_date($post_index) ?>
+                                <time class="post__time" datetime="<?=$post_time?>" title="<?=$post_time?>"><?=time_difference($post_time)?></time>
```
Формат даты/времени в атрибуте `datetime` и `title` должен быть разным

M2rk13 committed:
Доработки

Добавлено недостающее условие задания: title в теге time в другом формате даты

M2rk13 committed:
Merge branch 'module3-task2' into master

M2rk13 committed:
Merge pull request #6 from M2rk13/master

update module 3 task 2
